### PR TITLE
Proofread and crosslink examples with immunohistochemistry image.

### PR DESCRIPTION
**Pull request by mkcor**
## Description

These minor changes are part of the documentation effort, with a particular focus on the life sciences. Adding a link to another example helps users navigate the gallery.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: doc/examples/segmentation/plot_mask_slic.py
Comment (mkcor) [inline doc/examples/segmentation/plot_mask_slic.py:67]: Oops, I didn't catch this one. Anyhow.

Comment (rfezzani) [reply doc/examples/segmentation/plot_mask_slic.py:67]: Yes, :slightly_smiling_face:. For consistency, please apply the same changes here.

File: doc/examples/color_exposure/plot_ihc_color_separation.py
Comment (rfezzani) [inline doc/examples/color_exposure/plot_ihc_color_separation.py:64]: Is this change intentional?

**Review by rfezzani — COMMENTED** [review]
Thank you @mkcor, I just left a small question and a minor request :wink:

File: doc/examples/color_exposure/plot_ihc_color_separation.py
Comment (mkcor) [reply doc/examples/color_exposure/plot_ihc_color_separation.py:64]: Yes, I try to follow [The Canadian Style](https://www.btb.termiumplus.gc.ca/tcdnstyl-chap?lang=eng&lettr=chapsect7&info0=7.18) but maybe I didn't get this right?

File: doc/examples/segmentation/plot_mask_slic.py
Comment (mkcor) [reply doc/examples/segmentation/plot_mask_slic.py:67]: Sure. The reference here being https://github.com/scikit-image/scikit-image/pull/4850#discussion_r462005433 :wink:

File: doc/examples/color_exposure/plot_ihc_color_separation.py
Comment (rfezzani) [reply doc/examples/color_exposure/plot_ihc_color_separation.py:64]: As channels is not used between quotes [here](https://github.com/scikit-image/scikit-image/pull/5231/files/32ec5fcab4c5b296ceeb95a2138555adcad2ffe9#diff-3b977591f3df3413ae3f72b36779dfa715cd5a12058383a9a6e86f23c836364eR68), I think that we can remove them to avoid confusions
```suggestion
# Now we can easily manipulate the hematoxylin and DAB channels:
```

**Review by rfezzani — APPROVED** [review]
:tada: Thank you @mkcor :wink:

**Comment by mkcor** [comment]
Can another @scikit-image/core dev please review/approve this PR? Once it's merged, I can post my bio-centric announcement on the forum in all peace of mind. Thank you!

File: doc/examples/segmentation/plot_mask_slic.py
Comment (grlee77) [inline doc/examples/segmentation/plot_mask_slic.py:12]: I had to check the docs to see how this got rendered and notice "Separate" is capitalized, but I don't think it is a big issue if that is not easy to fix. 

Does it work if you use the following form?:

```
:ref:`sphx_glr_auto_examples_color_exposure_plot_ihc_color_separation.py <separate colors in immunohistochemical staining>`.
```

That would also be robust to future changes to the title in the other file. 

link to the rendered doc:
https://1426-2014929-gh.circle-artifacts.com/0/doc/build/html/auto_examples/segmentation/plot_mask_slic.html#sphx-glr-auto-examples-segmentation-plot-mask-slic-py

**Review by grlee77 — APPROVED** [review]
I left one small comment about possibly overriding cross-link text, but if that suggestion is not likely to work, feel free to merge as-is.

File: doc/examples/segmentation/plot_mask_slic.py
Comment (mkcor) [reply doc/examples/segmentation/plot_mask_slic.py:12]: Oh, yes. I expected this, for I used the same syntax before, in this example: https://scikit-image.org/docs/dev/auto_examples/applications/plot_3d_structure_tensor.html

I think it makes sense to display a proper title as opposed to a hyperlink within a sentence that should read smoothly... I admit I stayed between two stools in this case: I hesitated between "see tutorial: Title" and adapting the syntax to the example's title. Generally speaking, I'm in favour of using the imperative for example titles (just like commit messages), so there's a standard, plus it's engaging for the reader.

**Comment by grlee77** [comment]
Looks great, thanks @mkcor!

**Pull request merged**
